reunion/client/exchange_test.go

```diff
@@ -102,12 +109,14 @@ func TestClientServerBasics1(t *testing.T) {
 	go func() {
 		for {
 			update := <-aliceUpdateCh
+			require.NoError(update.Error)
 			if len(update.Result) > 0 {
 				aliceResult = update.Result
-				fmt.Printf("\n Alice got result: %s\n\n", update.Result)
-				wg.Done()
+				t.Log("Alice got result:", update.Result)
+				break
```

Comment:
agree this worker doesn't exit but doesn't updateCh receive more than one message?

Reply:
what can be done is to read from aliceUpdateCh like so:
update, ok := aliceUpdateCh
if !ok { // channel closed
  return
  }

Reply:
(and then ensure that the channel is closed at teardowN)